### PR TITLE
QTS: install function rewritten without using $wpdb

### PR DESCRIPTION
**Pull request by spleen1981**

### CODE REVIEW

File: modules/slugs/includes/class-qtranslate-slug.php [hunk 1]
```diff
@@ -108,17 +108,15 @@ public function get_meta_key( $force_lang = false ) {
...
     public function install() {
-        global $wpdb;
-
-        if ( function_exists( 'is_multisite' ) && is_multisite() ) {
-            if ( isset( $_GET['networkwide'] ) && ( $_GET['networkwide'] == 1 ) ) {
-                $old_blog = $wpdb->blogid;
-                $blogids  = $wpdb->get_col( $wpdb->prepare( "SELECT blog_id FROM $wpdb->blogs" ) );
-                foreach ( $blogids as $blog_id ) {
-                    switch_to_blog( $blog_id );
+        if ( is_multisite() ) {
```
Comment (herrvigg) [inline modules/slugs/includes/class-qtranslate-slug.php:114]: Not needed, as it's called in `is_plugin_active_for_network`.

File: modules/slugs/includes/class-qtranslate-slug.php [hunk 2]
```diff
@@ -108,17 +108,15 @@ public function get_meta_key( $force_lang = false ) {
...
+            if ( is_plugin_active_for_network(plugin_basename(QTRANSLATE_FILE)) ) {
+                $old_blog = get_current_blog_id();
+                $blogs  = wp_get_sites();
```
Comment (herrvigg) [inline modules/slugs/includes/class-qtranslate-slug.php:117]: Deprecated in WP 4.6.0 -> they say to use `get_sites`

File: modules/slugs/includes/class-qtranslate-slug.php [hunk 3]
```diff
@@ -108,17 +108,15 @@ public function get_meta_key( $force_lang = false ) {
...
+                foreach ( $blogs as $blog ) {
+                    switch_to_blog( $blog[blog_id]);
```
Comment (herrvigg) [inline modules/slugs/includes/class-qtranslate-slug.php:119]: `blog_id` is not defined. I suppose you meant `$blog->id`. Not tested apparently.

**Review by herrvigg — CHANGES_REQUESTED** [review]
Better with the WP API but there are a few things to fix.

File: modules/slugs/includes/class-qtranslate-slug.php (hunk 1 above)
Comment (spleen1981) [reply modules/slugs/includes/class-qtranslate-slug.php]: agree

File: modules/slugs/includes/class-qtranslate-slug.php (hunk 2 above)
Comment (spleen1981) [reply modules/slugs/includes/class-qtranslate-slug.php]: agree

File: modules/slugs/includes/class-qtranslate-slug.php (hunk 3 above)
Comment (spleen1981) [reply modules/slugs/includes/class-qtranslate-slug.php]: my bad, could not test multisite with my current test environment... this just looked like straight forward.
`wp_get_sites()` should return arrays with key `blog_id`, while `get_sites()` is returning `WP_Site` arrays with property `blog_id`. Agree and changed to the latter

**Review by herrvigg — APPROVED** [review]

**Pull request merged**